Add support for `--fb` option (like `--graphics` but without X)

intelfx opened the pull request:
Add support for a `--fb` option that starts QEMU in a graphical mode but without actually launching any graphical environment. This is intended for working with the framebuffer console (or anything else that might be run on top of such).

Additionally, this includes one change that synergize with the one above:
- the serial console is connected to stdio also when (any) graphics are used

This decouples the ability to inspect the kernel boot log or other console messages from the fact that the video output is (not) used. Previously, if a video output was used, the serial console on stdio was fully inhibited.

intelfx committed:
run: accept `--fb`, similar to `--graphics` but without starting X

Add support for a --fb option that starts QEMU in a graphical mode but
without actually launching any graphical environment. This is intended
for working with the framebuffer console (or anything else that might
be run on top of such).

Signed-off-by: Ivan Shapovalov <intelfx@intelfx.name>

intelfx committed:
run: dump kernel console on stdout even if graphics are used

Set up the primary console on hvc0 and connect it to stdio even when
graphics (either `--fb` or `--graphics`) are used.

The hvc0 console is configured as primary (`console=hvc0`), but an
additional cmdline parameter is passed (`virtme_console=tty1`) to
the init to keep the userspace session on the graphical console.

This decouples the ability to inspect the kernel boot log or other
console messages from the fact that the video output is (not) used.
Previously, if a video output of any kind was used, the serial console
on stdio was fully inhibited.

Signed-off-by: Ivan Shapovalov <intelfx@intelfx.name>

arighi reviewed (commented):
This looks interesting, do you have any special use case / example on how to use it? Also CI isn't happy, but should be an easy fix. Thanks!

matttbe reviewed (changes requested):
(Explicitly marking this PR as "Request Changes", because there are suggested changes.)

intelfx committed:
configkernel: set CONFIG_DRM_FBDEV_EMULATION=y for `virtme-run --fb`

Include `CONFIG_DRM_FBDEV_EMULATION=y` in the default kernel config,
making it possible to use fbcon (which is the intended use-case behind
the `virtme-run --fb` option).

Signed-off-by: Ivan Shapovalov <intelfx@intelfx.name>

matttbe reviewed (changes requested):
Thank you for the update!

One small comment for me. The rest looks OK, but I will let @arighi looking at it as he reviewed the first version.

matttbe reviewed (approved):
Thank you for the update!

As previously mentioned, I will let Andrea looking at it as he reviewed the first version.

matttbe commented:
@arighi: when you have the opportunity, do you mind checking this one, please? The modifications look good to me, but I didn't merge it because you commented a previous version.

arighi reviewed (approved):
Sorry for the late review. Everything looks good!

Side note: it also works together with `--systemd` which is pretty cool. Good job!
